### ISSUE TITLE
Different output when running on CUDA (compared to CPU and keras)

### ISSUE THREAD

**Issue by bazukas** [microsoft/onnxruntime#5798]
**Describe the bug**
When running our model with `CUDAExecutionProvider`, the output is a lot different, compared to `CPUExecutionProvider` (which is very similar to output of keras model).

Output on CPU:
```
In [38]: y_cpu[0][:10]
Out[38]:
array([ 4.0727073e-01, -8.2996595e-01,  7.8839634e-04, -7.4903473e-02,
       -2.8189439e-01,  3.4575549e-01, -9.9872723e-02, -1.3427838e+00,
       -7.6851761e-01, -5.5652875e-01], dtype=float32)
```
Output on CUDA:
```
In [39]: y_cuda[0][:10]
Out[39]:
array([-0.50394744, -0.51779425, -0.37129185, -0.84176487, -0.6755803 ,
       -1.3122077 , -0.13547337, -1.1968514 ,  0.8614615 , -0.21983027],
      dtype=float32)
```
Output on Keras:
```
In [40]: y_keras[0][:10]
Out[40]:
array([ 4.0726936e-01, -8.2997078e-01,  7.8449666e-04, -7.4905433e-02,
       -2.8189561e-01,  3.4576070e-01, -9.9873461e-02, -1.3427850e+00,
       -7.6851660e-01, -5.5653584e-01], dtype=float32)
```

**System information**
- Ubuntu 20.04
- `pip install onnxruntime-gpu`
- ONNX Runtime version: 1.4.0/1.5.2
- Python version: 3.7.6
- CUDA/cuDNN version: 10.1/10.2
- GPU model and memory: RTX1080ti

**To Reproduce**
Attaching the keras and onnx models as well as input example.
[Files](https://drive.google.com/file/d/1I4epVl1OA7jOzgR2GXWUV6qSeZjOPsdk/view?usp=sharing)

**Expected behavior**
Expect the outputs of the model to be very similar when running with CPU and CUDA execution providers.

**Comment by hariharans29** [comment]
Thanks. I will take a look.

**Comment by hariharans29** [comment]
Hi - Thanks for sharing the model. I have a feeling the reason for the diffs in the results of the CUDA EP when compared with the CPU EP is that the CUDA EP is not handling the auto_pad attribute in the pooling layers correctly (still running tests to confirm this)- 

![image](https://user-images.githubusercontent.com/9969784/99326012-4e2a5000-282c-11eb-8e1a-d3ab933e5074.png)

I ll confirm that this is the case and if this is the reason - I ll propose a fix for this.

**Comment by bazukas** [comment]
I'm not too sure, but the problem might be with the `keras2onnx` tool we used to convert to onnx.
When converting to tf format and then using the `tf2onnx` tool, we don't see the same problem anymore.

**Comment by hariharans29** [comment]
The model as is works correctly with the CPU EP right ? The CUDA EP produces incorrect results with the same model. So I wouldn't think this is a converter issue. This definitely warrants investigation and is most likely an ORT bug.

Each converter is nuanced and probably produces different model "flavors" . `tf2onnx` probably produces a model that doesn't trigger the bug in the CUDA EP, something which the model produces by `keras2onnx` does.

**Comment by gan3sh500** [comment]
Using tf2onnx after saving the model to savedmodel solves the issue.

**Comment by hariharans29** [comment]
I found the node causing the diffs - it is from the only BatchNorm node in the model. I ll update this when I send out a PR for the fix.

**Comment by snnn** [comment]
This issue has been automatically closed as 'not planned' because it has been marked as 'stale' for more than 30 days without activity. If you believe this is still an issue, please feel free to reopen it.
